### PR TITLE
Finance: check vault on initialization

### PR DESCRIPTION
**Pull request by sohkai**
> initialize check vault & etherToken != 0. Why not use kernel for vault & etherToken.

In the future, we could also use the kernel's default vault, if no vault is given.

TODO:

* [x] Wait until aragonOS is upgraded, and then rebase ontop of master
* [x] Use `IsContract` to check if the vault is a contract

Note that this is untestable due to the breaking Vault depedency :(.

### CODE REVIEW

File: apps/finance/contracts/Finance.sol [hunk 1]
```diff
@@ -119,6 +119,7 @@ contract Finance is AragonApp {
         initialized();
 
         require(_periodDuration > 1);
+        require(_vault != address(0));
```
Comment (bingen) [inline apps/finance/contracts/Finance.sol]: Maybe we could even check that `isContract`.

Comment (sohkai) [reply apps/finance/contracts/Finance.sol]: Was thinking the same, but even with `isContract` we can't be sure that it's a Vault... so not sure it's worth the extra gas. We're also not checking that in a few places (e.g. for the kernel in the app proxies).

Comment (bingen) [reply apps/finance/contracts/Finance.sol]: Well, I think the same reasoning would apply to several other places where we are using `isContract` and that even not making sure it is a Vault, at least we could avoid some more errors than only checking 0 address (so we increase safety although not up to 100%), and about the gas this function is called only once, but I'm fine leaving it this way.

Comment (sohkai) [reply apps/finance/contracts/Finance.sol]: Thinking about it some more, safer is better :). Will wait until `IsContract` is released in a new aragonOS.

File: apps/finance/contracts/Finance.sol [hunk 2]
```diff
@@ -15,7 +16,7 @@ import "@aragon/apps-vault/contracts/IVaultConnector.sol";
 import "@aragon/os/contracts/lib/misc/Migrations.sol";
 
 
-contract Finance is AragonApp {
+contract Finance is IsContract, AragonApp {
```
Comment (sohkai) [inline apps/finance/contracts/Finance.sol]: Note that solidity fails to linearize the inheritance tree if it's `AragonApp, IsContract`.

**Comment by bingen** [comment]
Incorporated into #425

**Pull request closed without merging**